Autopilot/Test/Src/Test_SensorFusion.cpp

```diff
@@ -0,0 +1,138 @@
+/*
+*
+* A skeleton for testing the sensor fusion algorithm
+*
+*/
+
+#include <gtest/gtest.h>
+#include "fff.h"
+
+#include "airspeed_Mock.hpp"
+#include "IMU_Mock.hpp"
+#include "SensorFusion.hpp"
+
+
+using namespace std;
+using ::testing::Test;
+using ::testing::_;
+using ::testing::SetArgReferee;
+
+/***********************************************************************************************************************
+ * Mocks
+ **********************************************************************************************************************/
+
+/***********************************************************************************************************************
+ * Variables
+ **********************************************************************************************************************/
+IMUData_t IMUTestData;
+airspeedData_t airspeedTestData;
+
+/***********************************************************************************************************************
+ * Tests
+ **********************************************************************************************************************/
+
+TEST(SensorFusion, FailedBusyIMUDataReturnsNegative1) {
+
+
+   	/***********************SETUP***********************/
+	
+	MockIMU imumock;
+	MockAirspeed airspeedmock;
+
+	IMUTestData.sensorStatus = -1;
+	airspeedTestData.sensorStatus = 0;
+	IMUTestData.isDataNew = 1;
+	airspeedTestData.isDataNew = 1;
+
+	SFError_t error;
+	SFOutput_t output;
+
+	/********************DEPENDENCIES*******************/
+	/********************STEPTHROUGH********************/
+	EXPECT_CALL(imumock, GetResult(_))
+		.WillOnce(DoAll(SetArgReferee<0>(IMUTestData)));
+
+	EXPECT_CALL(airspeedmock, GetResult(_))
+		.WillOnce(DoAll(SetArgReferee<0>(airspeedTestData)));
+	
+	error = SF_GetResult(&output, &imumock, &airspeedmock);	
+
+	/**********************ASSERTS**********************/
+
+	ASSERT_EQ(error.errorCode, -1);
+}
+
+TEST(SensorFusion, FailedBusyAirspeedDataReturnsNegative1 ) {
+
+   	/***********************SETUP***********************/
+	
+	MockIMU imumock;
+	MockAirspeed airspeedmock;
+
+	IMUTestData.sensorStatus = 0;
+	airspeedTestData.sensorStatus = -1;
+	IMUTestData.isDataNew = 1;
+	airspeedTestData.isDataNew = 1;
+
+	SFError_t error;
+	SFOutput_t output;
+
+	/********************DEPENDENCIES*******************/
+	/********************STEPTHROUGH********************/
+	EXPECT_CALL(imumock, GetResult(_))
```

Comment:
I had made that dependency section for the EXPECT_CALL stuff. I think things are much clearer if they are separated from the step through. Call it anything you like though.